src/main/java/pinger/Pinger.java

```diff
@@ -17,20 +17,25 @@ public static void main(String[] args)  {
                 System.out.println("Enter the URL to be pinged");
                 String urlI = "http://" + reader.readLine();
                 URL url = new URL(urlI);
-                HttpURLConnection conn = (HttpURLConnection) url.openConnection();
-                try (InputStream stream = conn.getInputStream()) {
-                    String content = readAllLines(stream);
-                    if (content.isEmpty()) {
-                        content = "No content returned";
+                URLConnection conn = url.openConnection();
+                if(conn instanceof HttpURLConnection) {
+                    try (InputStream stream = conn.getInputStream()) {
+                        String content = readAllLines(stream);
+                        if (content.isEmpty()) {
+                            content = "No content returned";
+                        }
+                        addField("Content Type", conn.getContentType());
```

Comment:
This doesn't compile.
You should make another variable, say `conn1`, and assign it to `(HttpURLConnection) conn`.